### DISCUSSION TITLE
Application cannot exit normally

### DISCUSSION THREAD

**Discussion by joezhouchenye** [winapps-org/winapps#536] [Support]
Winapps will continue running in the background even if all Windows application windows are closed.

This makes the Dash to Dock unclickable if other windows overlap with it. Killing the winapps process solves the issue.

![image](https://github.com/user-attachments/assets/ac45eba7-15f9-4576-9972-877d212ac649)

**Comment by KernelGhost** [comment]
@joezhouchenye It’s possible that the issue you’re experiencing stems from an [inherent limitation of the RDP protocol](https://techcommunity.microsoft.com/t5/security-compliance-and-identity/terminal-services-remoteapp-8482-session-termination-logic/ba-p/246566).

As noted in `bin/winapps`:
> RemoteApp RDP sessions require *AT LEAST* 20 seconds to be terminated by the Windows server.

This delay likely serves to improve the user experience by keeping the RDP session open briefly, preventing the need to re-establish an RDP connection if a user decides to open another application.

WinApps relies on FreeRDP to terminate first, and FreeRDP will only quit once the RDP session has been fully terminated by the Windows VM.

Have you tried waiting at least 20 seconds before running `ps -ef | grep winapps` to confirm the session has closed?

**Comment by WahPlus** [comment]
I have the same issue. Even after waiting 20 seconds, the session does not close. Moreover, when another app is opened during this, it does not show up. But if yet ANOTHER app is opened, THEN both show up under the name of the new app. Also, when the Windows session gets paused (libvirt), the process stops but the same double app start issue remains.

**Comment by jadegamesuk** [comment]
Also noticed this issue. It doesn't seem like it terminates the process at all. I've waited over 5 minutes on my machine and the process still runs in the background. Feel like I have to run the `ps -ef | grep winapps` command after I'm done to find and kill the process.

If it helps, running `winapps manual Winword.exe` in the terminal and closing Microsoft Word also has the same problem. It still runs in the background and needs manual intervention to close
